### PR TITLE
Moving HYPRE memory allocation macros to a common location [hypre-mem-macro-dev]

### PR DESCRIPTION
**Pull request by mlstowell**
This PR hopes to avoid code duplication when defining hyper memory allocation macros which are currently needed in two locations. There is at least one other PR #584 which will also need these definitions and there may be more in the future.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1172](https://github.com/mfem/mfem/pull/1172) | @mlstowell | @v-dobrev | @delyank + @v-dobrev | 04/29/20 | 05/06/20 | 05/13/20 | |

### CODE REVIEW

**Comment by v-dobrev** [comment]
The error in travis seems to be real. @mlstowell, can you double check?

**Comment by mlstowell** [comment]
> The error in travis seems to be real. @mlstowell, can you double check?

I took a quick look earlier and didn't find anything.  I've had a Valgrind job in the queue all afternoon but it hasn't started yet.  I'll keep an eye on it.

File: linalg/hypre_parcsr.hpp [hunk 1]
```diff
@@ -21,6 +21,56 @@
 
 #include "_hypre_parcsr_mv.h"
 
+// Define macro wrappers for hypre_TAlloc, hypre_CTAlloc and hypre_TFree:
+// mfem_hypre_TAlloc, mfem_hypre_CTAlloc, and mfem_hypre_TFree, respectively.
+// Note: the same macros are used in hypre.cpp, hypre_parcsr.cpp, and perhaps
```
Comment (v-dobrev) [inline linalg/hypre_parcsr.hpp:26]: "the same" -> "these"

**Comment by mlstowell** [comment]
> > The error in travis seems to be real. @mlstowell, can you double check?
> 
> I took a quick look earlier and didn't find anything. I've had a Valgrind job in the queue all afternoon but it hasn't started yet. I'll keep an eye on it.

I merged master into this branch last night and the tests now pass.

File: linalg/hypre_parcsr.hpp (hunk 1 above)
Comment (mlstowell) [reply linalg/hypre_parcsr.hpp]: Yes, that's much better.  I've made that change.

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
Merged in `next` for testing...

**Pull request merged**